docs/Project-Configuration/Task_Traces.md

```diff
@@ -18,6 +18,123 @@ When a task finishes Evergreen will check the task's [working directory](../Refe
+
+#### **Bad Examples:**
+- `status`
+- `identifier`
+- `function.name`
+- `function.status`
+
+#### **Good Examples:**
+- `task.status`
+- `version.status`
+- `project.identifier`
+- `resmoke.function.name`
+- `resmoke.function.status`
+
+By including a namespace (e.g., `resmoke`, `e2e`), attributes are contextualized, making their purpose within the system clear. This also simplifies data filtering and analysis. For instance:  
+- `task.status=success`  
+- `project.identifier=evergreen`
+
+---
+
+### **2. Avoid Excessive Dot Nesting**
+
+While namespaces are useful, over-nesting with too many dots can lead to overly complex attribute names. This makes them harder to read and analyze.
+
+#### **Bad Examples:**
+- `host.is.volume.migration`
+- `host.is.unexpirable`
+
+#### **Good Examples:**
+- `host.is_volume_migration`
+- `host.is_unexpirable`
+
+In these examples, underscores replace excessive dot nesting, keeping the attribute names concise and manageable.
+
+---
+
+### **3. Use Snake Case**
+
+When defining multi-word attributes, use `snake_case` to separate words. This ensures attributes are easy to read and understand.
+
+#### **Examples:**
+- **Bad:** `TestStatus`, `testStatus`, `test-status`  
+- **Good:** `test_status`, `test_duration`
+
+---
+
+### **4. Strive for Balance**
+
+Balance the need for context with simplicity. Use namespaces thoughtfully to provide clarity without overcomplicating attribute structures. This approach ensures readability and utility in analytics. Learn more about OpenTelemetry attribute naming [here](https://opentelemetry.io/docs/specs/semconv/general/attribute-naming/).
+
+---
+
+### **Span Names**
+
+Span names should be descriptive and concise, clearly describing the operation being performed.  
+#### **Examples:**  
+- **Bad:** `span1`, `operation_x`, `unknown_process`  
+- **Good:** `test`, `test_span`, `task_execution`
+
+---
+
+### **Attributes**
+
+Attributes provide additional context to a span. They should be:  
+- Key-value pairs (e.g., string, number, boolean).  
+- Descriptive, concise, and relevant.  
+- Consistent across spans to simplify search and analysis.
+
+#### **Examples:**
+- **Bad:**  
+  - Attribute names: `attribute1`, `attr2`  
+  - Lengthy names: `this_is_a_very_long_attribute_name_that_is_hard_to_read`  
+
+- **Good:**  
+  - `test_name="example_test"`  
+  - `test_status="success"`  
+  - `test_duration=123`
+
+**Consistency Tip:** Use the same attribute names across spans performing similar operations. For example, all spans representing tests should use `test_name` for the test name. Avoid inconsistent naming like `test_name` in one span and `test_case_name` in another.
+
+**Why It Matters:**  
+Inconsistent attribute naming can:  
+- Make it difficult to compare spans.  
+- Lead to exceeding Honeycomb's global limit of 10,000 attributes, causing trace drops.  
+
+---
+
+### **Events**
```

Comment:
Should we give guidance on [when to use an event vs an attribute](https://opentelemetry.io/docs/concepts/signals/traces/#when-to-use-span-events-versus-span-attributes)?